code/clue.js

```diff
@@ -1,59 +1,319 @@
+  mystery.weapon = randomSelector(weapons)
+  // This will change the styling and the content of the card after it is clicked
+  document.getElementById('weaponName').innerHTML = `type: ${mystery.weapon.name}`;
+  document.getElementById('weaponWeight').innerHTML = `weight: ${mystery.weapon.weight} kg`;
+  document.getElementById("start-view-card-weapon").style.display = "none";
+}
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+const loadingPickWeapon = () => {
+  // Sets time for loader animation and calls pickKiller function after
+  const loaderTime = setTimeout(pickWeapon, 2000);
+  // Hides the first "side"/ p-tag of the card
+  document.getElementById("start-view-card-weapon").style.display = "none";
+  // Hides the killer information
+  document.getElementById('weapon-info-wrapper').style.display = "none";
+  // Displays the loader
+  document.getElementById('loader-wrapper-weapon').style.display = "flex";
+  document.getElementById('weaponCard').style.backgroundColor = '#000'; 
 }
+// Calling the loadingPickWeapon function when clicking the card
+document.getElementById("weaponCard").onclick = loadingPickWeapon;
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//This function will be invoked in the loadingPickRoom function.
+const pickRoom = () => {
+  document.getElementById("loader-wrapper-room").style.display = "none";
+  document.getElementById("room-info-wrapper").style.display = "block";
+  // Randomly selects a room from the array. And add that to the mystery object.
+  mystery.room = randomSelector(rooms)
+  // This will change the styling and the content of the card after it is clicked
+  document.getElementById('roomName').innerHTML = `${mystery.room}`;
+  document.getElementById("start-view-card-room").style.display = "none";
+}
+
+const loadingPickRoom = () => {
+  // Sets time for loader animation and calls pickKiller function after
+  const loaderTime = setTimeout(pickRoom, 2000);
+  // Hides the first "side"/ p-tag of the card
+  document.getElementById("start-view-card-room").style.display = "none";
+  // Hides the killer information
+  document.getElementById('room-info-wrapper').style.display = "none";
+  // Displays the loader
+  document.getElementById('loader-wrapper-room').style.display = "flex";
+  document.getElementById('roomCard').style.backgroundColor = '#000'; 
+}
+// Calling the loadingPickRoom function when clicking the card
+document.getElementById("roomCard").onclick = loadingPickRoom;
+
+
+
+// Function to reveal mystery when clicking the button
+const revealMystery = () => {
+  if (mystery.killer === '') {
```

Comment:
Or we can simplify that : 

```
if (!mystery.killer) { ... }
```

For further reference check out https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/